File: Typescript/Algorithms/BinarySearch/BinarySearch.ts
export default function binarySearch(arr: number[], value: number): boolean {
    let low = 0;
    let high = arr.length;

    do {
        const median = Math.floor(low + (high - low) / 2);
        const arrValue = arr[median];
        if (arrValue === value)
            return true;
        else if (value > arrValue)
            low = median + 1;
        else
            high = median;
    } while (low < high);

    return false;
}

File: CSharp/Algorithms/BinarySearch/BinarySearch.cs
namespace CSharp.Algorithms.BinarySearch;

public class BinarySearch
{
    public static bool Search(int[] arr, int n)
    {
        int l = 0;
        int r = arr.Length;

        while (l < r)
        {
            int m = l + (int)Math.Floor((r - l) / 2.0);

            if (arr[m] > n)
                r = m - 1;
            else if (arr[m] < n)
                l = m + 1;
            else
                return true;
        }

        return false;
    }
}
